src/ffi/Timer.h

```diff
@@ -0,0 +1,57 @@
+// Copyright Microsoft and Project Verona Contributors.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <chrono>
+#include <string>
+#include <atomic>
+
+namespace verona::ffi
+{
+  /// Reports time elapsed between creation and destruction.
+  ///
+  /// Use:
+  ///  {
+  ///    auto T = TimeReport("My action");
+  ///    ... some action ...
+  ///  }
+  ///  // Here prints "My action: 12ms" to stderr
+  class TimeReport
+  {
+    timespec start;
+    std::string name;
+#ifdef CLOCK_PROF
+    static const clockid_t clock = CLOCK_PROF;
+#else
+    const clockid_t clock = CLOCK_PROCESS_CPUTIME_ID;
+#endif
+
+  public:
+    TimeReport(std::string n) : name(n)
+    {
+      std::atomic_signal_fence(std::memory_order::memory_order_seq_cst);
+      clock_gettime(clock, &start);
+    }
+    ~TimeReport()
+    {
+      using namespace std::chrono;
+      timespec end;
+      std::atomic_signal_fence(std::memory_order::memory_order_seq_cst);
+      clock_gettime(clock, &end);
+      std::atomic_signal_fence(std::memory_order::memory_order_seq_cst);
+      auto interval_from_timespec = [](timespec t) {
+        return seconds{t.tv_sec} + nanoseconds{t.tv_nsec};
+      };
+      auto elapsed =
+        interval_from_timespec(end) - interval_from_timespec(start);
+
+      // TODO: Avoid printfs on headers
```

Comment:
Not urgent, but all of this should probably be using `std::chrono` things and the existing flight recorder logic for logging (this is a copy-and-pasted version of a quick hack I wrote many years ago).

Reply:
Honestly, this is here mostly as legacy, as I'm not using it at all. I'd be happy to remove now. If we need something similar later, we can always rewrite it with `std::chrono` (as I have done a few times already and is trivial).

Reply:
If it's not used anywhere, removing it is probably a good idea.  It would be quite nice to have a generic thing like this: I found it very useful for seeing which bits of the clang logic were taking the most time.  It's not difficult to rewrite in the future if required.

Reply:
File removed